src/mon/OSDMonitor.cc

```diff
@@ -1367,8 +1367,7 @@ bool OSDMonitor::preprocess_query(MonOpRequestRef op)
     return preprocess_remove_snaps(op);
 
   default:
-    assert(0);
-    return true;
```

Comment:
I don't believe removing the return is safe or future proof. Granted, right now ceph is *always* compiled with assertions enabled, but that may not hold true in the future. This is not exactly redundant code - I take it as future proofing assertion handling, as at some point in time we may only compile assertions for debug/dev builds, and we must ensure the appropriate values are still returned.